plugins/system/cookiemanager/cookiemanager.php

```diff
@@ -0,0 +1,342 @@
+
+	/**
+	 * For cookie banner
+	 *
+	 * @var    CookieBanner
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $cookieBanner;
+
+	/**
+	 * For preferences banner
+	 *
+	 * @var    Preferences
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $preferences;
+
+	/**
+	 * Database object
+	 *
+	 * @var    DatabaseDriver
+	 * @since  __DEPLOY_VERSION__
+	 */
+	 protected $db;
+
+	 /**
+	  * For script in DOM
+	  *
+	  * @var    Script
+	  * @since  __DEPLOY_VERSION__
+	  */
+	 protected $script = [];
+
+	 /**
+	  * For scripts in DOM
+	  *
+	  * @var    Scripts
+	  * @since  __DEPLOY_VERSION__
+	  */
+	 protected $scripts;
+
+	 /**
+	  * For cookie category
+	  *
+	  * @var    Category
+	  * @since  __DEPLOY_VERSION__
+	  */
+	 protected $category;
+	/**
+	 * Add assets for the modal.
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+
+	 */
+	public function onBeforeCompileHead()
+	{
+		ob_start();
+		ob_implicit_flush(false);
+
+		if (!$this->app->isClient('site'))
+		{
+			return;
+		}
+
+		$this->app->getDocument()->getWebAssetManager()
+			->registerAndUseScript('plg_system_cookiemanager.script', 'plg_system_cookiemanager/cookiemanager.min.js', [], ['defer' => true], ['core'])
+			->registerAndUseStyle('plg_system_cookiemanager.style', 'plg_system_cookiemanager/cookiemanager.min.css');
+
+		$lang = Factory::getLanguage();
+		$lang->load('com_cookiemanager', JPATH_ADMINISTRATOR);
+
+		Text::script('COM_COOKIEMANAGER_PREFERENCES_LESS_BUTTON_TEXT');
+		Text::script('COM_COOKIEMANAGER_PREFERENCES_MORE_BUTTON_TEXT');
+
+		$params = ComponentHelper::getParams('com_cookiemanager');
+		$sitemenu = $this->app->getMenu();
+		$menuitem = $sitemenu->getItem($params->get('policylink'));
+
+		$config = [];
+		$config['expiration'] = $params->get('consent_expiration');
+		$config['position'] = $params->get('modal_position');
+		$this->app->getDocument()->addScriptOptions('config', $config);
+
+		$db    = $this->db;
+		$query = $db->getQuery(true)
+			->select($db->quoteName(['id','title','alias','description']))
+			->from($db->quoteName('#__categories'))
+			->where([
+				$db->quoteName('extension') . ' = ' . $db->quote('com_cookiemanager'),
+				$db->quoteName('published') . ' =  1',
+				]
+			)
+			->order($db->quoteName('lft'));
+
+		$db->setQuery($query);
+		$this->category = $db->loadObjectList();
+
+		$bannerBody = '<p>' . Text::_('COM_COOKIEMANAGER_COOKIE_BANNER_DESCRIPTION') . '</p><p><a '
+			. ' href="' . $menuitem->link . '">' . Text::_('COM_COOKIEMANAGER_VIEW_COOKIE_POLICY') . '</a></p>'
+			. '<h5>' . Text::_('COM_COOKIEMANAGER_MANAGE_CONSENT_PREFERENCES') . '</h5><ul>';
+
+		foreach ($this->category as $key => $value)
+		{
+			$bannerBody .= '<li class="cookie-cat form-check form-check-inline"><label>' . $value->title . '<span class="ms-4 form-check-inline form-switch"><input class="form-check-input" data-cookiecategory="'
+			. $value->alias . '" type=checkbox></span></label></li>';
+		}
+
+		$bannerBody .= '</ul>';

```

Comment:
Move HTML code to tmpl files, so they can be overriden.

Reply:
Done